integration_tests/src/lib.rs

```diff
@@ -24,6 +26,324 @@ mod test {
+
+                if !output.status.success() {
+                    bail!(
+                        "zpool create failed {} {}",
+                        String::from_utf8_lossy(&output.stdout),
+                        String::from_utf8_lossy(&output.stderr),
+                    );
+                }
+
+                // Create a dataset
+
+                let output = Command::new("pfexec")
+                    .args([
+                        "zfs",
+                        "create",
+                        &format!("{}/downstairs", pool_name),
+                    ])
+                    .output()?;
+
+                if !output.status.success() {
+                    let destroy_output = Command::new("pfexec")
+                        .args(["zpool", "destroy", &pool_name])
+                        .output()?;
+
+                    if !destroy_output.status.success() {
+                        panic!(
+                            "zpool destroy failed {} {}",
+                            String::from_utf8_lossy(&destroy_output.stdout),
+                            String::from_utf8_lossy(&destroy_output.stderr),
+                        );
+                    }
+
+                    bail!(
+                        "zfs create failed {} {}",
+                        String::from_utf8_lossy(&output.stdout),
+                        String::from_utf8_lossy(&output.stderr),
+                    );
+                }
+
+                // Adjust permissions
+
+                let dataset = TestDownstairsZfsDataset { tempdir, pool_name };
+
+                let uid: String = {
+                    let output = Command::new("id").args(["-u"]).output()?;
+
+                    if !output.status.success() {
+                        let destroy_output = Command::new("pfexec")
+                            .args(["zpool", "destroy", &dataset.pool_name])
+                            .output()?;
+
+                        if !destroy_output.status.success() {
+                            panic!(
+                                "zpool destroy failed {} {}",
+                                String::from_utf8_lossy(&destroy_output.stdout),
+                                String::from_utf8_lossy(&destroy_output.stderr),
+                            );
+                        }
+
+                        bail!(
+                            "id -u failed {} {}",
+                            String::from_utf8_lossy(&output.stdout),
+                            String::from_utf8_lossy(&output.stderr),
+                        );
+                    }
+
+                    String::from_utf8_lossy(&output.stdout).to_string()
+                };
+
+                let output = Command::new("pfexec")
+                    .args([
+                        "chown",
+                        uid.trim_end(), // remove '\n' from end
+                        &dataset.mountpoint()?,
+                    ])
+                    .output()?;
```

Comment:
If manually destroying the zpool is necessary (which I don't think is the case, see my comment [above](https://github.com/oxidecomputer/crucible/pull/1759/files#r2263041713)), then the `?` would be problematic, because it returns without destroying it.